package-best-practices.md

```diff
@@ -0,0 +1,41 @@
+
+Although the dbt Labs team completes a cursory review of new packages before adding them to the Package Hub, ultimately you are responsible for your package's performance.
+
+To avoid ambiguity, the bullets below are pretty formal, but we are a friendly bunch! If you need help preparing a package for deployment to the dbt Package Hub, reach out in [#package-ecosystem](https://getdbt.slack.com/archives/CU4MRJ7QB/) in the [dbt Community Slack](https://getdbt.com/community).
+
+---
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+### First run experience
+- Packages SHOULD contain a README file which explains how to get started with the package and customise its behaviour.
+
+### Customisability 
+- Packages MUST NOT hard-code table references, and MUST use `ref` or `source` instead.
+- Packages MAY provide a mechanism (such as variables) to enable/disable certain subsets of functionality.
+- Packages for data transformation:
+    - SHOULD provide a mechanism (such as variables) to customise the location of source tables.
+    - SHOULD NOT assume database/schema names in sources. 
+    - MAY assume table names, particularly if the package was built to support tables created by a known tool.
+
+### Dependencies
+- Packages requiring a minimum version of `dbt-core` MUST declare it in the `require-dbt-version` property of `dbt_project.yml`.
+- Packages requiring a minimum version of `dbt-core` SHOULD declare compatibility with all subsequent minor and patch releases of that major version. 
+    - For example, a package which depends on functionality added in dbt Core 1.2 SHOULD set its `require-dbt-version` property to `[">=1.2.0", "<2.0.0"]`.
+Packages with a dependency on another package:
+    - MAY, if the dependent package is major-version-zero, pin to the current minor version only.
+    - SHOULD NOT pin to a patch version of the dependent package unless they are aware of an incompatibility between their package and the dependency.
+    - SHOULD reference the version in the dbt Package Hub when available, as opposed to a `git` installation.
+
+### Interoperability
+- Packages MUST NOT override dbt Core behaviour in such a way as to impact other dbt resources (models, tests, etc) not provided by the package.
+- Packages SHOULD disambiguate their resource names to avoid clashes with nodes that are likely to already exist in a project. 
+    - For example, packages SHOULD NOT provide a model simply called `users`.
+
+### Updates
+- Packages SHOULD follow the guidance of the [Semantic Versioning Specification](https://semver.org/).
```

Comment:
```suggestion
### Releases and updates
- Packages SHOULD follow the guidance of the [Semantic Versioning Specification](https://semver.org/).
- Packages MUST create a git tag with the version number that follows the accepted version format.
```
We'd probably need to be more clear about what the "accepted version format" is -- currently it is _strictly_ https://semver.org/spec/v2.0.0.html but would be relaxed slightly if #130 is merged.